src/network/networkd-state-file.c

```diff
@@ -264,6 +271,8 @@ int manager_save(Manager *m) {
         online_state_str = link_online_state_to_string(online_state);
         if (online_state_str)
                 fprintf(f, "ONLINE_STATE=%s\n", online_state_str);
+        if (captive_portal)
+                fprintf(f, "CAPTIVE_PORTAL=%s\n", captive_portal);
```

Comment:
Why saved in the manager state??
That may be OK, but please also save it in link state (or, may be in dhcp lease file??).
Otherwise, sd_network_link_get_captive_portal() will never work.

Reply:
I didn't know where to put it honestly. I'll look at putting it in the lease state, that sounds correct.

Reply:
I settled on putting it in the link state.